Cesinrm

cesinrm opened the pull request:
Hi!

I've just added some new features.

delOne. Instead of flush, now, you can delete a route one by one.

getAll. Retrieve all routes.

add. Modified to accept 'verb' as part of the route.

I've also added a poor posibility to retrieve the HOWTO as html.
Guess you like it and sorry with commit's mess! It's my first github contribution and my first node.js, too!

Regards.

crodriguezVector committed:
Prueba de metodo para obtener todos los objetos que se han dado de alta.

crodriguezVector committed:
Añadir la ruta para que la reconozca como reservada

crodriguezVector committed:
Cambio a GET en getAll

crodriguezVector committed:
Añadir el README para que se puede conocer como usar el servidor

crodriguezVector committed:
Añadir un HOWTO para que se pueda buscar la info

crodriguezVector committed:
Corrección de cómo se accede al json del HOWTO

crodriguezVector committed:
no message

crodriguezVector committed:
Cambiar el README

crodriguezVector committed:
Modificaciones del README

crodriguezVector committed:
Modificaciones README

crodriguezVector committed:
Again README

crodriguezVector committed:
Prueba de Borrado individual

crodriguezVector committed:
Corrección del borrado

crodriguezVector committed:
More

crodriguezVector committed:
Commit

crodriguezVector committed:
Añadir el verbo al 'add' para que controle si esta recibiendo la peticion de verbo correcta

crodriguezVector committed:
logs

crodriguezVector committed:
cambiar igualdad de verbo

crodriguezVector committed:
came on

crodriguezVector committed:
Go, go, go

crodriguezVector committed:
Terminar el borrar un elemento

crodriguezVector committed:
logs

crodriguezVector committed:
fix delOne

crodriguezVector committed:
f

crodriguezVector committed:
probando borrado

crodriguezVector committed:
borrar logs innecesarios y finalizar el fixing del metodo "delOne"

crodriguezVector committed:
Comentar las nuevas opciones.

crodriguezVector committed:
Comentarios de HOWTO

crodriguezVector committed:
Probando el cambio del Howto a ruta fija

crodriguezVector committed:
Quitar el 200 OK

crodriguezVector committed:
Meter los headers correctos.

crodriguezVector committed:
Meter un README2.html para que se muestre como tal.

crodriguezVector committed:
Eliminar la ruta al HowTo como un sample json

crodriguezVector committed:
Delete personal data.

crodriguezVector committed:
Delete autorized for Flush

crodriguezVector committed:
Delete some comments.

crodriguezVector committed:
Delete unnecessary dependency

crodriguezVector committed:
Added tests for delOne and method matching

crodriguezVector committed:
Fix README.

bigodines commented on controller.js:
```diff
@@ -13,78 +13,136 @@ var argv = require('yargs').argv;
+			if (err) {
+				res.send(500, "FAKE-SERVER is misconfigured");
+			}
+			send(parseInt(200, 10), headers, data);
```
no need to parseInt if you know it is going to be an integer.

cesinrm replied:
Fixed.

bigodines commented on fakeresponse.js:
```diff
@@ -88,12 +92,34 @@ var FakeResponse = {
                 if(item.payload && !FakeResponse.matchRegex(item.payload, payload)) return false;
                 if(item.requiredHeaders && !FakeResponse.matchRegex(item.requiredHeaders, headers)) return false;
+                if(item.verb && !(item.verb==verb)) return false;
+                //Revisar para que sea dividido por el modulo en lugar de solo la primera vez que coincidan
```
comments in english, please.

cesinrm replied:
Deleted. It was unnecesary.

bigodines commented on fakeresponse.js:
```diff
@@ -88,12 +92,34 @@ var FakeResponse = {
     },
+    
+    /* Filters all items that match the URL and then tries to check if there is a specific behavior for the Nth call on the same endpoint */
+    matchDel: function (uri, responseCode, verb) {
```
this method is too similar to "match". Either we should merge them or make one of them use the other.

cesinrm replied:
I'll try to make a refactor with them.

cesinrm replied:
I did a refactor, merged both in one.

bigodines commented on test/controller.js:
```diff
@@ -301,4 +301,76 @@ describe('Integration tests', function () {
 
     });
+    
+    it('should provide a way to delete one response for a given endpoint', function () {
```
this test does not test the behaviour.. it just make sure we are calling the right method.

cesinrm replied:
You are right, I'm in the way to undestand some things in testing node.js with chai and others. Will try to improve my test.

cesinrm replied:
I've written some new test and fixed this one. Hope in the right way.

bigodines commented:
Thanks for your contribution. A few questions/requests tho:

- What is the rationale behind creating a README2.html and a /howTo endpoint?
- We use spaces and not "tabs".

all other comments are inline.

cesinrm commented:
You're welcome! 

@bigodines well, when I found your work throught github, I guessed that I could put it in my lan for my coworkers. So, README2.html, it's just a simple way to explain how it works for'em. 

Sorry for tabs!!! I forgot to change my editor configuration for js... :O

cesinrm committed:
Update fakeresponse.js

crodriguezVector committed:
Modifications:
- controller.js  calls to fakeresponse has changed
- fakeresponse.js refactorized match and delOne to be one method.
- test/controller.js improvements in test and changes
- test/fakeresponse.js improvements in test and changes

cesinrm commented:
I did some new changes with last commit. Hope everyhting is wright. 

Regards.

cesinrm commented:
I also have a change to add CORS if you want.

cesinrm commented:
Hi, @bigodines, I'm not sure if you think that I have to improve something. Don't hesitate to tell me. 

Regards.

bigodines commented:
unfortunately, I've lost permissions to merge your PR to this project as I don't work for yahoo anymore and have been removed from the organization :(

cesinrm commented:
:s well, I'll wait for the next master developer.

cesinrm committed:
Update README.md

yahoocla commented:
Thank you for submitting this pull request, however I do not see a valid CLA on file for you. Before we can merge this request please visit https://yahoocla.herokuapp.com/ and agree to the terms. Thanks! :smile:

cesinrm committed:
Update README.md

yahoocla commented:
CLA is valid!

cesinrm commented:
Anyone owns this repo?
